contracts/BluntDelegate.sol

```diff
@@ -148,9 +229,184 @@ contract BluntDelegate is IJBFundingCycleDataSource, IJBPayDelegate {
       _data.projectId != projectId
     ) revert INVALID_PAYMENT_EVENT();
 
-    // Make sure the token supply is under the cap.
-    if (hardCap != 0 && tokenStore.totalSupplyOf(_data.projectId) > hardCap) revert CAP_REACHED();
+    // Ensure contributed amount is a multiple of `TOKENS_PER_SLICE`
+    if (_data.amount.value % TOKENS_PER_SLICE != 0) revert VALUE_NOT_EXACT();
+
+    // Update totalContributions and contributions with amount paid
+    totalContributions += _data.amount.value;
 
-    // issue the slices to _data.beneficiary bassed on _data.amount.value.
+    // Make sure totalContributions is below `hardCap` and `MAX_CONTRIBUTION`
+    uint256 cap = hardCap != 0 && hardCap < MAX_CONTRIBUTION ? hardCap : MAX_CONTRIBUTION;
+    if (totalContributions > cap) revert CAP_REACHED();
+
+    // Cannot overflow as totalContributions would overflow first
+    unchecked {
+      contributions[_data.beneficiary] += _data.amount.value;
+    }
+  }
+
+  /**
+    @notice 
+    Part of IJBRedemptionDelegate, this function gets called when the beneficiary redeems tokens. It will update storage for the NFT mint.
+
+    @dev 
+    This function will revert if the contract calling is not one of the project's terminals. 
+    Value redeemed must be a multiple of 0.001 ETH.
+
+    @param _data The Juicebox standard project payment data.
+  */
+  function didRedeem(JBDidRedeemData calldata _data) external virtual override {
+    // Make sure the caller is a terminal of the project, and the call is being made on behalf of an interaction with the correct project.
+    if (
+      !directory.isTerminalOf(projectId, IJBPaymentTerminal(msg.sender)) ||
+      _data.projectId != projectId
+    ) revert INVALID_PAYMENT_EVENT();
+
+    // Ensure contributed amount is a multiple of `TOKENS_PER_SLICE`
+    if (_data.reclaimedAmount.value % TOKENS_PER_SLICE != 0) revert VALUE_NOT_EXACT();
+
+    // TODO: @jango Just to be sure, can we assume what's written in the line below to be always true?
+    // Cannot underflow as `_data.reclaimedAmount.value` cannot be higher than `contributions[_data.beneficiary]`
+    unchecked {
+      // Update totalContributions and contributions with amount redeemed
+      totalContributions -= _data.reclaimedAmount.value;
+      contributions[_data.beneficiary] -= _data.reclaimedAmount.value;
+    }
+  }
+
+  /**
+    @notice 
+    Creates slicer and issues `totalContributions` to this contract.
+
+    @dev 
+    This function will revert if the funding cycle related to the blunt round hasn't ended or if the slicer has already been created.
+  */
+  function issueSlices() external override {
+    if (slicerId != 0) revert SLICER_ALREADY_CREATED();
+
+    // TODO: @jango Add requirement: Revert if current funding cycle hasn't ended? What other requirements
+
+    // TODO: Add this in requirements for closing funding cycle
+    if (target != 0) {
+      if (totalContributions < target) revert TARGET_NOT_REACHED();
```

Comment:
Is the design goal here to dissolve the project if it hasn't met the requisite funding?

If so, it'd be neat if the project was owned by this contract, forcing a FC#2 (or n+1 if blunt round isn't the 1st cycle) that opens the treasury for redemptions. If the goal is met, the project ownership gets transferred to a pre-programmed address that can then make configuration decisions going forward.

Reply:
yes! perfect